Bottom: 223ce6d235f1094c7754992b576a426a840ea39b
Top:    84c99f10a4faca4be396d35fdd9250015616d30e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-04 18:24:17 +0000

Use strcasecmp when looking for the "outline" layer

A few cases used strcmp, leading to inconsistencies in our handling


---

diff --git a/src/draw.c b/src/draw.c
index f28f6c9..03c4335 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -901,8 +901,8 @@ DrawLayerGroup (int group, const BoxType * screen)
     {
       layernum = layers[i];
       Layer = PCB->Data->Layer + layers[i];
-      if (strcmp (Layer->Name, "outline") == 0
-	  || strcmp (Layer->Name, "route") == 0)
+      if (strcasecmp (Layer->Name, "outline") == 0
+	  || strcasecmp (Layer->Name, "route") == 0)
 	rv = 0;
       if (layernum < max_layer && Layer->On)
 	{
diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index 1f7946a..9c7e7e1 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -470,8 +470,8 @@ gerber_set_layer (const char *name, int group, int empty)
     return 0;
 
   flash_drills = 0;
-  if (strcmp (name, "outline") == 0
-      || strcmp (name, "route") == 0)
+  if (strcasecmp (name, "outline") == 0
+      || strcasecmp (name, "route") == 0)
     flash_drills = 1;
 
   if (is_drill && n_pending_drills)
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index edb7498..cc5c259 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1366,8 +1366,8 @@ DrawLayerGroup (int group, const BoxType * screen)
     layernum = layers[i];
     Layer = PCB->Data->Layer + layers[i];
 
-    if (strcmp (Layer->Name, "outline") == 0 ||
-        strcmp (Layer->Name, "route") == 0)
+    if (strcasecmp (Layer->Name, "outline") == 0 ||
+        strcasecmp (Layer->Name, "route") == 0)
       rv = 0;
 
     if (layernum < max_layer /*&& Layer->On*/) {
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index 0f1742f..b72fad9 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -380,8 +380,8 @@ ps_hid_export_to_file (FILE * the_file, HID_Attr_Val * options)
       if (layer->LineN || layer->TextN || layer->ArcN || layer->PolygonN)
 	print_group[GetLayerGroupNumberByNumber (i)] = 1;
 
-      if (strcmp (layer->Name, "outline") == 0
-	  || strcmp (layer->Name, "route") == 0)
+      if (strcasecmp (layer->Name, "outline") == 0
+	  || strcasecmp (layer->Name, "route") == 0)
 	{
 	  printf("see outline layer\n");
 	  outline_layer = layer;
@@ -717,8 +717,8 @@ ps_set_layer (const char *name, int group, int empty)
       && outline_layer
       && outline_layer != PCB->Data->Layer+idx
       && SL_TYPE (idx) == 0 /* copper */
-      && strcmp (name, "outline")
-      && strcmp (name, "route"))
+      && strcasecmp (name, "outline")
+      && strcasecmp (name, "route"))
     {
       printf("attempting to draw outlines on %s\n", name);
       DrawLayer (outline_layer, &region);
